lib/zypper.rb

```diff
@@ -27,32 +27,47 @@ class Zypper
 
   class <<self
     def isolated(options = {}, &block)
-      zypper_base = Dir.mktmpdir("machinery_zypper")
+      @zypp_base = Dir.mktmpdir("machinery_zypper")
```

Comment:
Since this is a class method you're essentially creating an instance variable on the class object itself. That's bad and not thread safe. I'd suggest to revert to passing the base directory path around as parameters again or make it an attribute of the zypper object instead.